.github/workflows/actions/publish-npm/action.yml

```diff
@@ -0,0 +1,38 @@
+name: 'Release'
+description: 'Releases Stencil Core'
+inputs:
+  version:
+    description: 'The type of version to release.'
+  tag:
+    description: 'The tag to publish to on NPM.'
+  token:
+    description: 'The NPM authentication token required to publish.'
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout Code
+      uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+    - name: Get Core Dependencies
+      uses: ./.github/workflows/actions/get-core-dependencies
+
+    - name: Download Build Archive
+      uses: ./.github/workflows/actions/download-archive
+      with:
+        name: stencil-core
+        path: .
+        filename: stencil-core-build.zip
+
+    - name: Set Version
+      run: npm version --no-git-tag-version ${{ inputs.version }}
+      shell: bash
+
+    - name: Prepare NPM Token
+      run: echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > .npmrc
+      shell: bash
+      env:
+        NPM_TOKEN: ${{ inputs.token }}
+
+    - name: Publish to NPM
+      run: npm publish --tag ${{ inputs.tag }}
+      shell: bash
```

Comment:
Is the idea for factoring out the publish step separately that we'll likely want to use this to publish things other than dev builds in the future? If so it makes sense to do so I think

Reply:
Yep - exactly. An example of this can be seen in the next PR in this train, where we use this step in a very similar way to publish nightly builds - https://github.com/ionic-team/stencil/pull/4100/files#diff-e3889abd7e85ba6d5d9dec9274d72a1ce3314fe8169ec90742fecefb40dbe554R55-R66